designer/server/src/main/scala/pl/touk/nussknacker/ui/statistics/StatisticsUrlsDomainService.scala

```diff
@@ -0,0 +1,89 @@
+package pl.touk.nussknacker.ui.statistics
+
+import cats.implicits.toTraverseOps
+import com.typesafe.scalalogging.LazyLogging
+import pl.touk.nussknacker.ui.statistics.StatisticsUrlsDomainService.{
+  emptyString,
+  nuStatsUrl,
+  queryParamsSeparator,
+  toURL,
+  urlBytesSizeLimit
+}
+
+import java.net.{URI, URL, URLEncoder}
+import java.nio.charset.StandardCharsets
+import scala.collection.mutable.ArrayBuffer
+import scala.util.{Failure, Success, Try}
+
+class StatisticsUrlsDomainService(private val fingerprint: Fingerprint, private val statistics: Map[String, String])
+    extends LazyLogging {
+
+  private lazy val queryParamsForEveryURL = List(
+    encodeQueryParam(NuFingerprint.name -> fingerprint.value)
+  )
+
+  def prepareUrls(): Either[StatisticError, List[URL]] = {
+    val JoinedQueryParams = statistics.toList
+      // Sorting for purpose of easier testing
+      .sortBy(_._1)
+      .map(encodeQueryParam)
+    splitUrlsByBytesSizeLimit(JoinedQueryParams)
+      .map(toURL)
+      .sequence
+  }
+
+  private def encodeQueryParam(entry: (String, String)): String =
+    s"${URLEncoder.encode(entry._1, StandardCharsets.UTF_8)}=${URLEncoder.encode(entry._2, StandardCharsets.UTF_8)}"
+
+  private def splitUrlsByBytesSizeLimit(queryParamsWithSize: List[String]): List[String] = {
+    val (_, queryParamsAcc, urlsAcc) = queryParamsWithSize
+      .foldLeft((0, ArrayBuffer[String](), ArrayBuffer[Option[String]]())) {
```

Comment:
I wonder if we shouldn't have a util extension to group strings in chunks with a max size, eg:
```scala
implicit class Chunked(val strings: Iterable[String]) extends AnyVal {

  def groupByMaxChunkSize(maxChunkSizeInBytes: Int): List[List[String]]
}
```

The URL creation logic won't be affected by the chunk creation logic.
WDYT?